cordapp/src/main/kotlin/net/corda/plugins/CordappPlugin.kt

```diff
@@ -100,33 +106,79 @@ class CordappPlugin : Plugin<Project> {
         jarTask.dependsOn(task)
     }
 
+    private fun isCordappProject(project: Project): Boolean{
```

Comment:
The `isCordappProject()` function bothers me, because it looks like something that shouldn't need to exist, i.e. it's asking "Have I applied me to myself?".

By applying the `cordapp` plugin, we should be _telling_ Gradle that _this_ project's POM _only_ needs to be handled differently. Remember that Corda contains multiple `cordapp` modules - `:finance:contracts` and `:finance:workflows` for instance - which means that the `apply` logic will be executed multiple times too.

Reply:
It is applied multiple times, and you are right - this is worrying - but I literally cannot find a way of hooking into the pom task in a better way. 

And yes, this is to guard against it being applied to generateMavenPom tasks outside of the current project.

Reply:
I'll investigate along the lines of what the Shadow plugin does:
https://github.com/johnrengelman/shadow/blob/bdf2b01df548e961d2ed202ad4214305f65d1c7e/src/main/groovy/com/github/jengelman/gradle/plugins/shadow/ShadowJavaPlugin.groovy#L66